### PR TITLE
users: fix nil panic

### PR DESCRIPTION
**Pull request by mrnugget**
The `LogPasswordEvent` function can be called with `nil` as the request. This fixes the nil panic from this call: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/database/users.go?L1531:3

It also removes the deafult values that don't need to be set to remove references to the field.

The regression was introduced in https://github.com/sourcegraph/sourcegraph/pull/49174

## Test plan

- Existing tests and manual testing

### CODE REVIEW

File: internal/database/users.go [hunk 1]
```diff
@@ -1566,12 +1562,14 @@ func LogPasswordEvent(ctx context.Context, db DB, r *http.Request, name Security
...
 		Name:            string(name),
-		URL:             r.URL.Host,
 		AnonymousUserID: "backend",
 		Argument:        eArgs,
 		Source:          "BACKEND",
 		Timestamp:       time.Now(),
 	}
+	if r != nil && r.URL != nil {
+		logEvent.URL = r.URL.Host
+	}
```
Comment (kopancek) [inline internal/database/users.go:1572]: We already make a check for `r != nil` on line 1545, might be better to extend it there with:
```
	var path string
	var host string
	if r != nil {
		path = r.URL.Path
		host = r.URL.Host
	}
```

**Review by kopancek — APPROVED** [review]

File: internal/database/users.go (hunk 1 above)
Comment (mrnugget) [reply internal/database/users.go:1572]: will do!

**Review by kopancek — APPROVED** [review]

File: internal/database/users.go (hunk 1 above)
Comment (nathan-downs) [reply internal/database/users.go]: @mrnugget the thought behind it was that the security event never leaves the instance, so having the path is fine, but the URL in the event will be passed along to the analytics database, so we only want the host so we’re not exposing private repo names. Added the host in for parity with other events we pass along, but it’s also probably fine to take out altogether.

Appreciate the fix!

**Review by sashaostrikov — COMMENTED** [review]
nice fix!

**Pull request merged**